docs/GettingStartedAriesDev/PlugIns.md

```diff
@@ -0,0 +1,172 @@
+    Startup->>+ArgParse: configure
+    ArgParse->>settings:  ["external_plugins"]
+    ArgParse->>settings:  ["blocked_plugins"]
+
+    Note right of Startup: Each configured plug-in is validated and loaded
+    Startup->>+DefaultContext:  build_context()
+      DefaultContext->>DefaultContext:  load_plugins()
+      DefaultContext->>+PluginRegistry:  register_package() (for built-in protocols)
+        PluginRegistry->>PluginRegistry:  register_plugin() (for each sub-package)
+      DefaultContext->>PluginRegistry:  register_plugin() (for non-protocol built-ins)
+      loop for each external plug-in
+	    DefaultContext->>PluginRegistry:  register_plugin()
+	    alt if a setup method is provided
+	      PluginRegistry->>ExternalPlugIn:  has setup
+	    else if routes and/or message_types are provided
+	      PluginRegistry->>ExternalPlugIn:  has routes
+	      PluginRegistry->>ExternalPlugIn:  has message_types
+	    end
+	    opt if definition is provided
+	      PluginRegistry->>ExternalPlugIn:  definition()
+	    end
+      end
+      DefaultContext->>PluginRegistry:  init_context()
+        loop for each external plug-in
+	      alt if a setup method is provided
+	        PluginRegistry->>ExternalPlugIn:  setup()
+	      else if a setup method is NOT provided
+	        PluginRegistry->>PluginRegistry:  load_protocols()
+	        PluginRegistry->>PluginRegistry:  load_protocol_version()
+	        PluginRegistry->>ProtocolRegistry:  register_message_types()
+	        PluginRegistry->>ProtocolRegistry:  register_controllers()
+	      end
+	      PluginRegistry->>PluginRegistry:  register_protocol_events()
+	    end
+
+	Note right of Startup: If the admin server is enabled, plug-in routes are added
+	Startup->>AdminServer:  create admin server if enabled
+	Startup->>AdminServer:  setup_context() (called on each request)
+	  AdminServer->>PluginRegistry:  register_admin_routes()
+	  loop for each external plug-in
+	    PluginRegistry->>ExternalPlugIn:  routes.register() (to register endpoints)
+	  end
+```
+
+## Developing a New Plug-In
+
+When developing a new plug-in:
+
+- If you are providing a new protocol or defining message types, you *should* include a `definition.py` file.
+- If you are providing a new protocol or defining message types, you *should* include a `message_types.py` file.
+- If you are providing additional Admin endpoints, you *should* include a `routes.py` file.
+- If you are providing any other functionality, you should provide a `setup.py` file to initialize the custom functionality.  No guidance is *currently* available for this option.
+
+### PIP vs Poetry Support
+
+Most Aca-Py plug-ins provide support for installing the plug-in using [poetry](https://python-poetry.org/).  It is *recommended* to include support in your package for installing using *either* pip or poetry, to provide maximum support for users of your plug-in.
+
+### Plug-In Demo
+
+TBD
```

Comment:
I like this idea... maybe illustrate all the ways mentioned above in the "When developing a new plug-in" section.  Or we can begin pointing at specific implementations that exist in the wild.

Reply:
Yes that's what I was thinking.  I'm going to create a new issue for this